mlflow/pyspark/optuna/storage.py

```diff
@@ -0,0 +1,414 @@
+        run = self._mlflow_client.get_run(study_id)
+        directions_str = run.data.tags["optuna.study_direction"]
+        directions = [StudyDirection[name] for name in directions_str.split(",")]
+        return directions
+
+    def get_study_user_attrs(self, study_id) -> Dict[str, Any]:
+        run = self._mlflow_client.get_run(study_id)
+        user_attrs = {}
+        for key, value in run.data.tags.items():
+            if key.startswith("user_"):
+                user_attrs[key[5:]] = json.loads(value)
+        return user_attrs
+
+    def get_study_system_attrs(self, study_id) -> Dict[str, Any]:
+        run = self._mlflow_client.get_run(study_id)
+        system_attrs = {}
+        for key, value in run.data.tags.items():
+            if key.startswith("sys_"):
+                system_attrs[key[4:]] = json.loads(value)
+        return system_attrs
+
+    def get_all_studies(self) -> List[FrozenStudy]:
+        # print("Getting all studies")
+        runs = self._mlflow_client.search_runs(experiment_ids=[self._experiment_id])
+        studies = []
+        for run in runs:
+            study_id = run.info.run_id
+            study_name = run.data.tags["mlflow.runName"]
+            directions_str = run.data.tags["optuna.study_direction"]
+            directions = [StudyDirection[name] for name in directions_str.split(",")]
+            studies.append(
+                FrozenStudy(
+                    study_name=study_name,
+                    direction=None,
+                    directions=directions,
+                    user_attrs=self.get_study_user_attrs(study_id),
+                    system_attrs=self.get_study_system_attrs(study_id),
+                    study_id=study_id,
+                ))
+        return studies
+
+    def create_new_trial(self, study_id, template_trial: Optional[FrozenTrial] = None) -> int:
+        # print(f"Creating new trial for study ID: {study_id}")
+        if template_trial:
+            frozen = copy.deepcopy(template_trial)
+        else:
+            frozen = FrozenTrial(
+                trial_id=-1,  # dummy value.
+                number=-1,  # dummy value.
+                state=TrialState.RUNNING,
+                params={},
+                distributions={},
+                user_attrs={},
+                system_attrs={},
+                value=None,
+                intermediate_values={},
+                datetime_start=datetime.datetime.now(),
+                datetime_complete=None,
+            )
+
+        distribution_json = {
+            k: distribution_to_json(dist)
+            for k, dist in frozen.distributions.items()
+        }
+        distribution_str = json.dumps(distribution_json)
+        tags = {"param_directions": distribution_str}
+
+        trial_run = self._mlflow_client.create_run(experiment_id=self._experiment_id, tags=tags)
+        trial_id = trial_run.info.run_id
+        self._mlflow_client.set_tag(trial_id, MLFLOW_PARENT_RUN_ID, study_id)
```

Comment:
Can this be included in the tags definition in line 209 to limit the API call frequency?